src/server/survey_and_program_data.R

```diff
@@ -61,6 +61,27 @@ programDataHeaders <- list(tot= "Total Tests",
                             ancpos= "Total Positive ANC Tests"
 )
 
+validateProgramData <- function(df) {
+
+    if (is.null(df))
+        return(FALSE)
+
+    validateYear <- function(givenYear) {
+
+        rows <- df[!is.na(df$year) & df$year == givenYear,]
+
+        if (nrow(rows) == 0) return(TRUE)
```

Comment:
I found the series of ifs a little hard to follow on first looking, though I don't know if there is a neat way to make it look cleaner in R. There is a case_when function in dplyr which may make it a little easier to read. What do you think?

```
dplyr::case_when(
          nrow(rows) == 0 ~ TRUE,
          nrow(rows) == 1 ~ identical(as.character(rows$sex),c("both")),
          nrow(rows) == 2 ~ identical(sort(as.character(rows$sex)), sort(c("male", "female"))),
          TRUE ~ FALSE )
```

The `TRUE ~ FALSE` bit is not very neat.